src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -92,7 +93,7 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, AccessDeniedException {
```

Comment:
This implementation will still cause the program to crash if the file becomes read-only. Think about why this might be the case.